src/Sulu/Component/HttpCache/Handler/PublicHandler.php

```diff
@@ -59,6 +59,11 @@ public function updateResponse(Response $response, StructureInterface $structure
             return;
         }
 
+        // when structure cache-lifetime disabled - return
+        if ((int) $structure->getCacheLifeTime() === 0) {
+            return;
+        }
```

Comment:
0 should disable the cache for now. And yes we should have a look a the `$maxAge` and `$sharedMaxAge`. Also I'm not happy with the Event-Listener. It often leads unpredictable behavior.